### PR TITLE
Switch to ajv validator

### PR DESCRIPTION
**Pull request by sgentry**
### Reasons for making this change

Switching to the [ajv](https://github.com/epoberezkin/ajv) validator should help address validation problems outlined in these issues:

#384 
#702 

### Checklist

* [ ] **I'm updating documentation**
  - [ ] I've checked the rendering of the Markdown text I've added
  - [ ] If I'm adding a new section, I've updated the Table of Content
* [x ] **I'm adding or updating code**
  - [x ] I've added and/or updated tests
  - [ ] I've updated docs if needed
  - [ x] I've run `npm run cs-format` on my branch to conform my code to [prettier](https://github.com/prettier/prettier) coding style
* [ ] **I'm adding a new feature**
  - [ ] I've updated the playground with an example use of the feature

### CODE REVIEW

File: webpack.config.prod.js [hunk 1]
```diff
@@ -39,6 +39,15 @@ module.exports = {
...
         ],
+      },
+      {
+        test: /\.json$/,
+        loader:"json-loader",
+        include: [
+          path.join(__dirname, "css"),
+          path.join(__dirname, "playground"),
+          path.join(__dirname, "node_modules"),
+        ],
```
Comment (glasserc) [inline webpack.config.prod.js]: What's up with these json-loader changes? Do we import JSON explicitly anywhere in the project? I don't see any new uses of this functionality as part of this PR..

File: src/validate.js [hunk 2]
```diff
@@ -99,6 +112,31 @@ function unwrapErrorHandler(errorHandler) {
...
+ */
+function transformAjvErrors(errors = []) {
+  if (errors === null) {
+    return [];
+  }
+
+  return errors.map(e => {
+    const { dataPath, keyword, message, params } = e;
+    // not having 'instance' in the property name breaks things further down the line.
+    let property = `instance${dataPath}`;
```
Comment (glasserc) [inline src/validate.js]: Could you elaborate on this?

File: webpack.config.prod.js (hunk 1 above)
Comment (sgentry) [reply webpack.config.prod.js]: Ajv loads two json files so I added the loader for that.

ajv/lib/refs/json-schema-draft-06.json
ajv/lib/refs/$data.json

It uses draft 6 by default but we can specify an earlier revision if needed. Not sure if using v6 would be a breaking change for this lib.

File: src/validate.js (hunk 2 above)
Comment (sgentry) [reply src/validate.js]: Not having 'instance' in the property value caused an additional 8 tests to fail so I was thinking that there's a dependency in some of the components for this, and didn't dig further. Maybe I just need to update the tests. I'll take a closer look at the tests.

File: webpack.config.prod.js (hunk 1 above)
Comment (glasserc) [reply webpack.config.prod.js]: Ah, great, thanks. I think draft 6 is what we target at the moment.

File: src/validate.js (hunk 2 above)
Comment (sgentry) [reply src/validate.js]: I pushed another commit to remove the reliance on `instance` and updated the affected tests.

**Comment by sgentry** [comment]
I agree that switching from `arguments` -> `params` could be a breaking change for anyone who may use it. I guess another option is to try and re-create `arguments` based on the ajv error list but I don't think I'd like to attempt that.

**Comment by glasserc** [comment]
Thank you very much for this great work!

**Pull request merged**